eng/azpbuild.sh

```diff
@@ -0,0 +1,42 @@
+source="${BASH_SOURCE[0]}"
+
+# resolve $source until the file is no longer a symlink
+while [[ -h "$source" ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+
+repoRoot="$( cd "$scriptroot/.." && pwd )"
+tmpDir="$repoRoot/artifacts/tmp"
+
+if [ ! -d $tmpDir ]; then
+    # 'mkdir -p' isn't technically guaranteed to be available on all POSIX systems
+    # So if this fails, we just need to mk each dir separately.
+    mkdir -p $tmpDir
+fi
+
+loggerRsp="$tmpDir/logger.rsp"
+loggersZip="$tmpDir/msbuildlogger.zip"
+loggerAssemblyName="Microsoft.TeamFoundation.DistributedTask.MSBuild.Logger.dll"
+loggerAssembly="$tmpDir/$loggerAssemblyName"
+
+# Until we figure out a better way to get this...
+# This the same URL used by the Azure tasks: https://github.com/Microsoft/azure-pipelines-tasks/blob/da57c7efbf858a628786c869d7d14b887d1d90d7/Tasks/Common/MSBuildHelpers/make.json#L5
+# And it's HTTPS, so I think we can trust it.
+echo "Downloading Azure Pipelines MSBuild logger..."
+curl -sSL https://vstsagenttools.blob.core.windows.net/tools/msbuildlogger/3/msbuildlogger.zip "-o$loggersZip" >/dev/null
```

Comment:
There should be a space between `-o` and the output filename.

The other option is to `cd` to the `$tmpDir` and use `-O`. That works because the filename in the URL matches what you want on disk.

Reply:
I'd rather use `-o` so we fix the file name and know what it is for later.

I definitely had `-o [filename]` at first and `curl` errored. I'll try it again though.

Reply:
The space is certainly not _required_ though it certainly is my stylistic preference.